### PR TITLE
Refactor DRF JSON:API

### PR DESCRIPTION
**Pull request by SandyRogers**
This PR:
- removes EMG fork of DRFJA, to use latest version of upstream instead.
- adds various workarounds with EMG API, especially in our JSON and CSV renderers, to cater for using custom fields (accessions and lineages etc) instead of `pk` fields.
- fixes many cases of broken relationships and `?include=`. Things should be (more) JSON:API compliant now.
- update Django templates to EMBL VF2 look-and-feel
- tidies up a few things with API Docs generation.

Still TODO (in different PRs):
- Fix warnings and errors with Open API Spec Generation by Spectacular. Some are quite involved, due to the many overrides we do on e.g. serializers using `self.context['request']`, which is not present for the schema generation inspection.
- Improve content of view docstrings, especially adding analysis examples.
- More library upgrades, once we can drop Python3.6 (this is needed for the latest versions of Django, DRF, DRFJA etc)

### CODE REVIEW

**Review by gustavo-salazar — APPROVED** [review]
Happy with the changes.
Great effort @SandyRogers 

I reckon once merged, we put this for a week or 2 in wwwdev and wwwint and play around the website to make sure we didn't miss anything.

File: emgui/templates/rest_framework/login_base.html [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+{% load rest_framework %}
 
 {% block content %}
-    {% include "rest_framework/login_form.html" %}
-{% endblock %}
+    <h3>MGnify login</h3>
+    <form action="{% url 'rest_framework:login' %}" role="form" method="post" class="vf-stack vf-stack--400">
+      <div class="vf-form__item vf-stack">
+
+        <label for="text" class="vf-form__label vf-form__label--required">
```
Comment (mberacochea) [inline emgui/templates/rest_framework/login_base.html:9]: Is this compatible with the current website styling?.

Comment (SandyRogers) [reply emgui/templates/rest_framework/login_base.html:9]: Everything UI in this PR is VF2 style, so will look a bit out-of-place with the stable website, but okay with the beta website. So perhaps makes sense to release this to prod at the same time as making the frontend a public beta.

File: emgapi/models.py [hunk 2]
```diff
@@ -307,7 +307,7 @@ class Meta:
         db_table = 'BIOME_HIERARCHY_TREE'
         ordering = ('biome_id',)
         unique_together = (
-            ('biome_id', 'biome_name'),
+            ('lineage', 'biome_name'),
```
Comment (mberacochea) [inline emgapi/models.py:310]: Don't we need a migration for this change?

File: emgcli/settings.py [hunk 3]
```diff
@@ -358,14 +358,14 @@ def create_secret_key(var_dir):
     ),
 
     'DEFAULT_RENDERER_CLASSES': (
-        'rest_framework_json_api.renderers.JSONRenderer',
+        # 'rest_framework_json_api.renderers.JSONRenderer',
         'emgapi.renderers.DefaultJSONRenderer',
```
Comment (mberacochea) [inline emgcli/settings.py:362]: Can you please remove the commented renderers? I've meaning to do that for a long time

**Review by mberacochea — APPROVED** [review]
Brilliant, great stuff Sandy!.

**Pull request merged**